### DISCUSSION TITLE
Remote adapter on 2nd Raspberry - no connection - missing permissions?

### DISCUSSION THREAD

**Discussion by Queequack** [Koenkk/zigbee2mqtt#11951] [Devices]
Hi, 
In advance: In the past I have successfully used the CC2531 stick on the same Pi (server). But now I have to restructure… 
I want to set up a ZigBee coordinator on a second Raspberry that is not the MQTT broker. Zigbee2MQTT with ser2net enables the setup of a remote adapter (ser2net links the serial port to a TCP port that can be accessed over LAN).

**Hardware _[and location]_:**
- Pi1: Raspberry Pi 4, Server with openHABian (headless) and Mosquitto MQTT Broker _[located in basement]_
- Pi2: Raspberry Pi 4, Client with Raspberry OS (with desktop) _[located centrally in main floor]_, with Texas Instruments CC2531 USB Stick (Note: The CC2531 stick is not ideal. However, the problem remains the same with another stick)

**Software:**
1. Mosquitto MQTT Broker on **Pi1** (tested successfully)
2. Zibee2MQTT on **Pi1** (installed according to [this](https://www.zigbee2mqtt.io/guide/installation/01_linux.html#configuring))
   - configuration.yaml: Changed the port `/dev/ttyACM0` to `'tcp://192.168.0.41:20108'` which is the static IP and the defined port of Pi2. 
3. ser2net (version 4.3.3-1) on **Pi2** (installed according to [this](https://www.zigbee2mqtt.io/advanced/remote-adapter/connect_to_a_remote_adapter.html#_1-install-ser2net) (2b for version >= 4.0) 
 
**Error:** See logs below. I also checked the [Zigbee2MQTT fails to start section](https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html).

`ls -l /dev/serial/by-id.` 
on **Pi2** delivers the correct serial port: `lrwxrwxrwx 1 root root 13 25. Mär 09:33 usb-Texas_Instruments_TI_CC2531_USB_CDC___0X00124B0014D9611A-if00 -> ../../ttyACM0`

I suspect **P1** does not have permissions to access the port. But I do not know how to set this up. I have also tried all possible spellings of the tcp address…
`test -w tcp://192.168.0.41:20108 && echo success || echo failure`
on **Pi1** delivers always a `failure`. Whereas it works on Pi2 if I use `/dev/ttyACM0` as port (delivers `success`).

If I can get this to work, there's one more question. Probably I will get it working with trial and error... How exactly do I enter the port in openHAB? before `/dev/ttyACM0`. With TCP tunnel `tcp://192.168.0.41:20108` or `‘tcp://192.168.0.41:20108‘` or `192.168.0.41:20108` or `192.168.0.41,20108`…

**Logs:**
Starting ZigBee with
`cd /opt/zigbee2mqtt`
`npm start`

delivers:
pi@pi40:~ $ cd /opt/zigbee2mqtt
pi@pi40:/opt/zigbee2mqtt $ npm start

zigbee2mqtt@1.24.0 start /opt/zigbee2mqtt
node index.js

Zigbee2MQTT:info  2022-03-25 09:02:52: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2022-03-25.09-02-48' filename: log.txt
Zigbee2MQTT:info  2022-03-25 09:02:53: Starting Zigbee2MQTT version 1.24.0 (commit #a73caa3e)
Zigbee2MQTT:info  2022-03-25 09:02:53: Starting zigbee-herdsman (0.14.20)
Zigbee2MQTT:error 2022-03-25 09:02:54: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-03-25 09:02:54: Failed to start zigbee
Zigbee2MQTT:error 2022-03-25 09:02:54: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-03-25 09:02:54: Exiting...
Zigbee2MQTT:error 2022-03-25 09:02:55: Error: Error while opening socket
    at Socket.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/znp/znp.ts:198:24)
    at Socket.emit (events.js:412:35)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.24.0 start: 'node index.js'
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the zigbee2mqtt@1.24.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2022-03-25T08_02_55_162Z-debug.log

**Content of log file:**
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'start' ]
2 info using npm@6.14.16
3 info using node@v14.19.1
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle zigbee2mqtt@1.24.0 prestart: zigbee2mqtt@1.24.0
6 info lifecycle zigbee2mqtt@1.24.0 start: zigbee2mqtt@1.24.0
7 verbose lifecycle zigbee2mqtt@1.24.0 start: unsafe-perm in lifecycle true
8 verbose lifecycle zigbee2mqtt@1.24.0 start: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/opt/zigbee2mqtt/node_modules/.bin:/usr/l>
9 verbose lifecycle zigbee2mqtt@1.24.0 start: CWD: /opt/zigbee2mqtt
10 silly lifecycle zigbee2mqtt@1.24.0 start: Args: [ '-c', 'node index.js' ]
11 silly lifecycle zigbee2mqtt@1.24.0 start: Returned: code: 1  signal: null
12 info lifecycle zigbee2mqtt@1.24.0 start: Failed to exec start script
13 verbose stack Error: zigbee2mqtt@1.24.0 start: node index.js
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:400:28)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:400:28)
13 verbose stack     at maybeClose (internal/child_process.js:1058:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)
14 verbose pkgid zigbee2mqtt@1.24.0
15 verbose cwd /opt/zigbee2mqtt
16 verbose Linux 5.10.103-v7l+
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "start"
18 verbose node v14.19.1
19 verbose npm  v6.14.16
20 error code ELIFECYCLE
21 error errno 1
22 error zigbee2mqtt@1.24.0 start: node index.js
22 error Exit status 1
23 error Failed at the zigbee2mqtt@1.24.0 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by hocomadvies** [comment]
Did you ever solved this? I'm in the same process trying to get the stick working through a second pi. Getting the same error that the socket is not opening.. So I am curious if you found a solution here? Might be something as opening a port or so?

**Reply by Queequack** [reply]
Hi, unfortunately I haven't solved it yet. I haven't set up the Zigbee network yet. However, it will come to me soon. I might give it another try. If I can't get it working, I may set up a separate gateway somewhere.
